ambari-server/src/main/java/org/apache/ambari/server/state/UpgradeHelper.java

```diff
@@ -1013,6 +1013,9 @@ private void processConfigurationsIfRequired(UpgradeContext upgradeContext)
           String configurationType = currentServiceConfig.getType();
 
           Config currentClusterConfigForService = cluster.getDesiredConfigByType(configurationType);
+          if (currentClusterConfigForService == null) {
```

Comment:
Could this be a legitimate scenario if an upgraded component is adding a new config type?  If so, then we probably do not want to throw an exception.

Reply:
We are adding new config types later in the code. The code under question goes through existing configs, and they are supposed to have a selected version (we even have a check for that that is triggered on Ambari startup).

Reply:
thanks for the clarification.